File: easy/python/168.py
# https://leetcode.com/problems/excel-sheet-column-title/
# Time: O(log(n))
# Space: O(1)
class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        answer = []

        while columnNumber > 0:
            columnNumber -= 1
            char = chr((columnNumber % 26) + 65)
            answer.append(char)
            columnNumber = columnNumber // 26
        answer.reverse()

        return "".join(answer)

File: easy/ts/168.ts
// https://leetcode.com/problems/excel-sheet-column-title/
// Time: O(log(n))
// Space: O(1)
function convertToTitle(columnNumber: number): string {
  let ans: string[] = [];

  while (columnNumber > 0) {
    columnNumber--;
    let charValue = String.fromCharCode((columnNumber % 26) + 65);
    ans.push(charValue);
    columnNumber = Math.floor(columnNumber / 26);
  }
  return ans.reverse().join("");
}
